File: app/src/main/java/com/example/admin/proyectotesttesis/JsonDataDetectedSaved/JsonDetection.java
package com.example.admin.proyectotesttesis.JsonDataDetectedSaved;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.LinkedList;
import java.util.List;

public class JsonDetection {

    private JSONObject jsonDetections = null;
    private List<BoxDetected> boxDetectedList = null;

    public JsonDetection( List<BoxDetected> boxDetectedList ){
        this.boxDetectedList = boxDetectedList;
        this.jsonDetections = new JSONObject();
    }

    public void addBoxDetected( BoxDetected boxDetected ) {
        if( this.boxDetectedList == null ) { this.boxDetectedList = new LinkedList<BoxDetected>(); }
        this.boxDetectedList.add(boxDetected);
    }

    public void makeJsonDetections(){
        if ( this.jsonDetections== null ){ this.jsonDetections = new JSONObject(); }

        for ( BoxDetected boxDetected : this.boxDetectedList ){
            try {
                JSONObject jsonImg = new JSONObject();
                jsonImg.put("name", boxDetected.getNameImg());


                if( boxDetected.hasRegions() ){
                    JSONObject jsonBoxes = new JSONObject();
                    int i = 0;
                    for (Region region : boxDetected.getRegions()) {
                        JSONObject jsonBox = new JSONObject();
                        jsonBox.put("xmin", region.getX1());
                        jsonBox.put("ymin", region.getY1());
                        jsonBox.put("xmax", region.getX2());
                        jsonBox.put("ymax", region.getY2());
                        jsonBox.put("width", region.getWidth());
                        jsonBox.put("height", region.getHeight());

                        jsonBoxes.put(""+i,jsonBox);
                        i++;
                    }
                    jsonImg.put("Boxes", jsonBoxes);
                }

                this.jsonDetections.put(boxDetected.getNameImg().split("\\.")[0], jsonImg);
            }catch (Exception e){
                e.printStackTrace();
            }
        }
    }

    @Override
    public String toString() {
        return this.jsonDetections.toString();
    }
}

File: app/src/main/java/com/example/admin/proyectotesttesis/JsonDataDetectedSaved/JsonDetection.kt
package com.example.admin.proyectotesttesis.JsonDataDetectedSaved

import org.json.JSONArray
import org.json.JSONObject
import java.util.*

class JsonDetection( boxDetectedList: List<BoxDetected> ){
    private var jsonDetections: JSONObject?
    private lateinit  var boxDetectedList: List<BoxDetected>


    init {
        this.jsonDetections = JSONObject()
    }

    fun addBoxDetected( boxDetected: BoxDetected ) {
        if( this.boxDetectedList == null ) { this.boxDetectedList = LinkedList<BoxDetected>() }

        (this.boxDetectedList as LinkedList<BoxDetected> ).add(boxDetected)
    }

    fun saveDetections(){
        if ( this.jsonDetections== null ){ this.jsonDetections = JSONObject() }

        for ( boxDetected: BoxDetected in this.boxDetectedList ){
            var jsonBox = JSONObject()
            jsonBox.put("filename", boxDetected.nameImg)

            var jsonArrRegion = JSONArray()
            for ( region:Region in boxDetected.regions ){
                var jsonRegion = JSONObject()
                jsonRegion.put("x1", region.x1)
                jsonRegion.put("y1", region.y1)
                jsonRegion.put("x2", region.x2)
                jsonRegion.put("y2", region.y2)
                jsonRegion.put("witdth", region.width)
                jsonRegion.put("height", region.height)

                jsonArrRegion.put( jsonRegion )
            }
            jsonBox.put("Regions", jsonArrRegion)
            this.jsonDetections?.put(""+boxDetected.nameImg, jsonBox)
        }


    }

    override fun toString(): String {
        return this.jsonDetections.toString()
    }

}